src/java/org/apache/cassandra/index/sai/disk/v1/IndexSearcher.java

```diff
@@ -126,4 +126,6 @@ RangeIterator<Long> toSSTableRowIdsIterator(PostingList postingList, QueryContex
 
         return new SSTableRowIdsRangeIterator(indexContext, searcherContext);
     }
+
+    public abstract PostingList searchPosting(QueryContext context, Expression exp, AbstractBounds<PartitionPosition> keyRange, int limit) throws IOException;
```

Comment:
why is this added to IndexSearcher?

Reply:
I needed to call `VIS.searchPosting` on an instance of `IndexSearcher` where the `IndexContext` had a `VectorType`; so, I needed to add the `searchPosting` abstract method in the `IndexSearcher` class itself, which the `VIS` extended.